### PR TITLE
STO Jira integration DOC-2491

### PR DESCRIPTION
**Pull request by douglas-j-bothwell**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

**Review by aklaft — APPROVED** [review]
looks good to me

File: docs/security-testing-orchestration/use-sto/jira-integrations.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+* This feature is behind the Feature Flag `STO_JIRA_INTEGRATION`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
+:::
+
+## Set up the Jira integration
+
+1. Create an API key for your Jira account. For more information, go to [Manage API tokens for your Atlassian account](https://support.atlassian.com/atlassian-account/docs/manage-api-tokens-for-your-atlassian-account/) in the Atlassian documentation.
+
+2. Create a Jira connector as described in [Connect to Jira](/docs/platform/7_Connectors/connect-to-jira). Note the following:
+    * Create your connector at the Account level.
+    * Create a [text secret](/docs/platform/security/add-use-text-secrets/) for your Atlassiant API key.
```
Comment (chris-harness) [inline docs/security-testing-orchestration/use-sto/jira-integrations.md:30]: This secret MUST use the Harness Secret Manager (not an external secret store like Vault)

File: docs/security-testing-orchestration/use-sto/jira-integrations.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+* This feature is not supported in Self-Managed Enterprise Edition.
+* This feature is behind the Feature Flag `STO_JIRA_INTEGRATION`. Contact [Harness Support](mailto:support@harness.io) to enable the feature.
+:::
+
+## Set up the Jira integration
+
+1. Create an API key for your Jira account. For more information, go to [Manage API tokens for your Atlassian account](https://support.atlassian.com/atlassian-account/docs/manage-api-tokens-for-your-atlassian-account/) in the Atlassian documentation.
+
+2. Create a Jira connector as described in [Connect to Jira](/docs/platform/7_Connectors/connect-to-jira). Note the following:
+    * Create your connector at the Account level.
```
Comment (chris-harness) [inline docs/security-testing-orchestration/use-sto/jira-integrations.md:29]: There will soon be support for using connectors from the org or project level

File: docs/security-testing-orchestration/use-sto/jira-integrations.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+1. Go to the **Pipeline Execution** &gt; **Security Tests** page with the detected issue. 
+
+2. Click on the issue to select it. Then click **Create Ticket**. 
+
+   ![](./static/jira-integration-01-create-ticket.png)
+
+3. In **Create Ticket in Jira**, set the options as follows:
+
+    1. **Scope** Create a separate ticket for each target in the project (**This target**), or create one ticket for the issue across all targets in the project (**This project**).
```
Comment (chris-harness) [inline docs/security-testing-orchestration/use-sto/jira-integrations.md:54]:
```suggestion
    1. **Scope** Create a ticket for all occurrences of the issue in the current target (**This target**), or create a ticket for all occurrences of the issue across all targets in the project (**This project**).
```

File: docs/security-testing-orchestration/use-sto/jira-integrations.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+    2. **Jira Project** The Jira project for the ticket. The Jira connector populates this list based on the projects in your Jira account.
```
Comment (chris-harness) [inline docs/security-testing-orchestration/use-sto/jira-integrations.md:56]:
```suggestion
    2. **Jira Project** The Jira project for the ticket. The drop-down will be populated with a list of all Jira projects available via the Jira Connector created earlier.
```

File: docs/security-testing-orchestration/use-sto/jira-integrations.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+    3. **Issue Type** The Jira ticket type.
```
Comment (chris-harness) [inline docs/security-testing-orchestration/use-sto/jira-integrations.md:58]:
```suggestion
    3. **Issue Type** The Jira issue type for the ticket. 
```

File: docs/security-testing-orchestration/use-sto/jira-integrations.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+    1. **Scope** Create a separate ticket for each target in the project (**This target**), or create one ticket for the issue across all targets in the project (**This project**).
+
+    2. **Jira Project** The Jira project for the ticket. The Jira connector populates this list based on the projects in your Jira account.
+
+    3. **Issue Type** The Jira ticket type. 
+
+    4. **Title** The Jira ticket title. 
+
+    5. **Notes** Any notes you want to add to the ticket.
```
Comment (chris-harness) [inline docs/security-testing-orchestration/use-sto/jira-integrations.md:62]:
```suggestion
    4. **Title** The Jira issue title. 

    5. **Notes** Any notes you want to add to the Jira issue. 
```

File: docs/security-testing-orchestration/use-sto/jira-integrations.md [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+<img src={create_ticket} alt="Create Jira Ticket options" height="50%" width="50%" />
+```
+-->
+
+4. Click **Create**. The Jira connector sends the request to the Jira API and Jira creates the ticket. 
+
+
+Note the following: 
+
+   * In the Issue Details pane, the **Create Ticket** button is replace with the Jira ticket number.
```
Comment (chris-harness) [inline docs/security-testing-orchestration/use-sto/jira-integrations.md:77]:
```suggestion
   * In the Issue Details pane, the **Create Ticket** button is now replaced with the Jira issue number. The current status of the issue in Jira is show.
```

File: docs/security-testing-orchestration/use-sto/jira-integrations.md [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+     
+     ![](./static/jira-integration-03-ticket-num-in-issue-details.png)
+      <!--
```
Comment (chris-harness) [inline docs/security-testing-orchestration/use-sto/jira-integrations.md:80]: Screenshot is outdated

File: docs/security-testing-orchestration/use-sto/jira-integrations.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+    2. **Jira Project** The Jira project for the ticket. The Jira connector populates this list based on the projects in your Jira account.
+
+    3. **Issue Type** The Jira ticket type. 
+
+    4. **Title** The Jira ticket title. 
+
+    5. **Notes** Any notes you want to add to the ticket. 
+
+       ![](./static/jira-integration-02-create-ticket.png)
+
```
Comment (chris-harness) [inline docs/security-testing-orchestration/use-sto/jira-integrations.md:65]: Screenshot is outdated

**Comment by chris-harness** [comment]
@douglas-j-bothwell I'm not sure how to best resolve the naming of Jira Issue (which is what a ticket is called in Jira) vs. the abstract concept of an External Ticket on the Harness platform (which will eventually include additional ticketing system providers). We also need to avoid the confusion of Security *Issue* with Jira *Issue*

**Comment by douglas-j-bothwell** [comment]
@chris-harness WRT the use of "Jira issue" vs. "Jira ticket:"  The current UI uses "Ticket" so I'm going use this term in the docs to keep things consistent. If we want to standardize on the term "Jira issue," we should update the UI as well as the doc.

**Comment by douglas-j-bothwell** [comment]
I've implemented @chris-harness edits in a separate pull request. Doc is live: https://developer.harness.io/docs/security-testing-orchestration/use-sto/jira-integrations

**Pull request closed without merging**